src/pages/profile/index.js

```diff
@@ -1,7 +1,264 @@
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import Avatar from '../../components/avatar';
+import { get, patch } from '../../service/apiClient';
+import useAuth from '../../hooks/useAuth';
+import jwtDecode from 'jwt-decode';
+import Form from '../../components/form';
+import TextInput from '../../components/form/textInput';
+
+import './styles.css';
+
 const ProfilePage = () => {
+  const { id } = useParams();
+  const [profileUser, setProfileUser] = useState(null);
+  const [loggedInUser, setLoggedInUser] = useState(null);
+  const { token } = useAuth();
+  const [canEdit, setCanEdit] = useState(false);
+  const [formData, setFormData] = useState(null);
+  const [prevProfileUserId, setPrevProfileUserId] = useState(null); // New state variable
+
+  useEffect(() => {
+    const fetchUserData = async (profileId, setter) => {
+      try {
+        const response = await get(`users/${profileId}`);
+        setter(response.data);
+      } catch (error) {
+        console.error(`Error fetching user data for ID ${profileId}:`, error);
+      }
+    };
+
+    if (token) {
+      const loggedInId = jwtDecode(token).userId;
```

Comment:
you should be able to get this from the auth context -> the logged in user is available from the auth context, so there should not be a need to re-fetch the data?

Reply:
The logged in user gets set to null when i reload the page, which is why i rely on fetching every time.